PropTechAPI - コピー/src/main/java/com/PropTech/UserRegist/PropTechAppAPI/PropTechAPI/Logic/UserRegistrationLogic.java

```diff
@@ -0,0 +1,213 @@
+package com.PropTech.UserRegist.PropTechAppAPI.PropTechAPI.Logic;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.util.Date;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+
+import com.PropTech.UserRegist.PropTechApp.PropTechAPI.PasswordHashUtil.PasswordHashUtil;
+import com.PropTech.UserRegist.PropTechApp.PropTechAPI.Response.ResponseCommon;
+import com.PropTech.UserRegist.PropTechApp.PropTechAPI.Response.ResponseDetails;
+import com.PropTech.UserRegist.PropTechAppAPI.PropTechAPI.EntityDTO.Request;
+import com.PropTech.UserRegist.PropTechAppAPI.PropTechAPI.EntityDTO.RequestBody;
+import com.PropTech.UserRegist.PropTechAppAPI.PropTechAPI.Util.StringUtil;
+
+@RequestScoped
+public class UserRegistrationLogic {
+	
+	//@Inject
+	StringUtil stringutil = new StringUtil();
+	
+	Request request = new Request();
+	
+	RequestBody body = new RequestBody();
+	
+	public UserRegistrationLogic() {
+		
+	}
+	
+	private final static String URL = "jdbc:postgresql://localhost:5432/UserRegistrationAPI";
+	private final static String USER = "postgres";
+	private final static String PASSWORD = "password";
+	
+	public Response GenerateResponse(ResponseCommon responsecommon) {
+		ResponseDetails responsedetails = new ResponseDetails();
+		responsedetails.setTime(LocalDateTime.now().toString());
+		responsedetails.setCode(responsecommon.getStatus());
+		responsedetails.setMessage(responsecommon.getResponseMessage());
+		return Response.status(responsecommon.getStatus()).entity(responsedetails).build();
+	}
+	
+	public Response CRUDService(Request request, RequestBody body) {
+		ResponseDetails responsedetails = new ResponseDetails();
+		ResponseCommon responsecommon = ResponseCommon.S201;
+		StringBuilder seqID = new StringBuilder();
+		StringBuilder HashPassword = new StringBuilder();
+		
+		Response TypeUserResponse = RequestBodyCheck(request, body);

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *TypeUserResponse*. The resource is returned at line 57. There are other execution paths that don&#39;t close the resource or return it, for example, when an else condition at line 56 evaluates to true. Ensure *TypeUserResponse* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).